.github/workflows/codeql.yml

```diff
@@ -12,12 +12,12 @@
 name: 'CodeQL'
 
 on:
-  push:
-    branches: ['dev']
   pull_request:
     # The branches below must be a subset of the branches above
     branches: ['dev']
```

Comment:
This needs to include the `main` branch otherwise a required CI job is not being executed. 

Similar for MegaLinter too.